TaskManagementSystem/main.go

```diff
@@ -7,27 +7,58 @@ import (
 	"time"
 )
 
-func session2() {
-	var ProjectStatus string = "in progress"
+func TaskUpdate(CompletedTasks int) {
+
 	const NumberOfTotalTasks int = 100
-	CompletedTasks := 25
-	const (
-		Low = iota
-		Medium
-		High
-	)
-	var ProjectCompletionStatus bool = true
+	var ProjectStatus string
+
+	RemainingTasks := NumberOfTotalTasks - CompletedTasks
+	if CompletedTasks != NumberOfTotalTasks {
+		ProjectStatus = "in progress"
+	} else {
+		ProjectStatus = "finished"
+	}
+
+	var ProjectCompletionStatus bool
 	if ProjectStatus == "in progress" {
 		ProjectCompletionStatus = false
 	} else {
 		ProjectCompletionStatus = true
 	}
-	fmt.Println("Current project status:", strings.ToUpper(ProjectStatus))
-	fmt.Printf("Tasks completed: %d out of %d\n", CompletedTasks, NumberOfTotalTasks)
+
+	const (
+		Low = iota + 1
+		Medium
+		High
+	)
+
+	var ProjectPhase string
+	switch {
+	case CompletedTasks < 30:
+		ProjectPhase = "Starting phase"
+	case 30 <= CompletedTasks && CompletedTasks <= 60:
+		ProjectPhase = "Midway"
+	case 60 <= CompletedTasks:
+		ProjectPhase = "Final phase"
+	default:
+		fmt.Println("Wrong entered data")
+	}
+
+	fmt.Printf("Tasks remaining %d out of %d\n", RemainingTasks, NumberOfTotalTasks)
+	fmt.Printf("Current project status: %s\n", strings.ToUpper(ProjectStatus))
+	fmt.Printf("Project is in the %s.\n", strings.ToLower(ProjectPhase))
```

Comment:
strings.ToUpper() function is costly. Will be good static data set as should be. ToUpper() and likely other functions usually uses for data validation cases. Try to not make over-engineering, this will make your application compile and run slowly, eat more unusual computer resources.